src/Tasks/CreateItem.cs

```diff
@@ -162,10 +162,9 @@ private static ITaskItem[] ExpandWildcards(ITaskItem[] expand)
                                 expanded.Add(newItem);
                             }
                         }
-                        catch (DriveEnumerationWildcardException ex)
+                        catch (DriveEnumerationWildcardException)
```

Comment:
How should this exception be caught later and converted into an `InvalidProjectFileException`?

Reply:
Insert a catch wherever you want it caught and converted like:

```C#
catch (DriveEnumerationWildcardException e)
{
    throw new InvalidProjectFileException(..., e);
}

Reply:
This would mean that I need to either add the following in Microsoft.Build.Tasks.csproj & Microsoft.Build.Utilities.csproj:
```
    <Compile Include="..\Build\Errors\InvalidProjectFileException.cs">
      <ExcludeFromStyleCop>true</ExcludeFromStyleCop>
    </Compile>
```

Or add the exception to shared code (and add the shared code to both .csproj files). Initially, I was hoping to add `ProjectErrorUtilities.ThrowInvalidProject(...)` here, but that would require internals to be exposed and made public.

Reply:
I don't think the first would work. You'd have "different" InvalidProjectFileExceptions. You can make internals visible to other assemblies without making them public—have you checked to make sure that isn't already true?

Reply:
This is possible - see https://github.com/dotnet/msbuild/pull/7029#discussion_r746828081, but it is still necessary to add Microsoft.Build.csproj to Microsoft.Tasks.Build.csproj as a `ProjectReference`, which caused many errors/warnings. I'm not certain if it is okay to remove many of the <Compile Include> statements in Microsoft.Tasks.Build.csproj to resolve this.

Reply:
Yes and no. It's actually good to get rid of as many of those unnecessary double compilations as you can. The risk is that if someone wants to load (in this case) the Tasks assembly without loading Microsoft.Build, that's no longer viable if their code goes down any of the relevant code paths. I'm not super confident in all the use cases in which you might want to, but I think most of the things Tasks rely on live in Framework and not Build, which means people who want to use one of our Tasks without using the rest of MSBuild would no longer be able to. I could be wrong about that if, for instance, all Tasks actually need Build, which also sounds very possible.